py-polars/polars/expr/expr.py

```diff
@@ -3783,16 +3778,37 @@ def apply(
         │ 1   ┆ c   ┆ 2         │
         └─────┴─────┴───────────┘
 
-        It is better to implement this with an expression:
+        Tip: it is better to implement this with an expression:
 
         >>> df.with_columns(
         ...     (pl.col("a") * 2).alias("a_times_2"),
         ... )  # doctest: +IGNORE_RESULT
 
-        In a GroupBy context the function is applied by group:
+        In a GroupBy context, each element of the Series is itself a Series:
+
+        >>> df.lazy().groupby("b").agg(pl.col("a")).sort("b").collect()
+        shape: (3, 2)
+        ┌─────┬───────────┐
+        │ b   ┆ a         │
+        │ --- ┆ ---       │
+        │ str ┆ list[i64] │
+        ╞═════╪═══════════╡
+        │ a   ┆ [1]       │
+        │ b   ┆ [2]       │
+        │ c   ┆ [3, 1]    │
+        └─────┴───────────┘
+        >>> df.lazy().groupby("b").agg(pl.col("a")).sort("b").collect()["a"][-1]
+        shape: (2,)
+        Series: '' [i64]
+        [
+                3
+                1
+        ]
+
+        Therefore, from the user's point-of-view, the function is applied by group:
 
-        >>> df.lazy().groupby("b", maintain_order=True).agg(
-        ...     pl.col("a").apply(lambda x: x.sum())
+        >>> df.lazy().groupby("b").agg(pl.col("a").apply(lambda x: x.sum())).sort(
+        ...     "b"
```

Comment:
minor drive-by - I think it would be better to not teach `maintain_order=True`, as that's more expensive. the example can be made deterministic with `sort`, and then at least it should be clearer that that part is optional, and people will default to the more performant `maintain_order=False`

Reply:
I agree we shouldn't be including `maintain_order` in every groupby example. However, adding `sort` everywhere adds additional clutter that I'd rather not have in the examples.

I'd vote to simply add a `# doctest: +IGNORE_RESULT` to these examples.